lib/core/src/sdk.rs

```diff
@@ -3022,8 +3062,60 @@ impl LiquidSdk {
         &self,
         req: PrepareLnUrlPayRequest,
     ) -> Result<PrepareLnUrlPayResponse, LnUrlPayError> {
+        let amount_msat = match req.amount {
+            PayAmount::Drain => {
+                let get_info_res = self
+                    .get_info()
+                    .await
+                    .map_err(|e| LnUrlPayError::Generic { err: e.to_string() })?;
+                ensure_sdk!(
+                    get_info_res.wallet_info.pending_receive_sat == 0
+                        && get_info_res.wallet_info.pending_send_sat == 0,
+                    LnUrlPayError::Generic {
+                        err: "Cannot drain while there are pending payments".to_string(),
+                    }
+                );
+                let lbtc_pair = self
+                    .swapper
+                    .get_submarine_pairs()?
+                    .ok_or(PaymentError::PairsNotFound)?;
+                let drain_fees_sat = self.estimate_drain_tx_fee(None, None).await?;
+                let drain_amount_sat = get_info_res.wallet_info.balance_sat - drain_fees_sat;
+                // Get the inverse invoice amount by calculating a dummy amount then increment up to the drain amount
+                let dummy_fees_sat = lbtc_pair.fees.total(drain_amount_sat);
+                let dummy_amount_sat = drain_amount_sat - dummy_fees_sat;
+                let invoice_amount_sat = utils::increment_invoice_amount_up_to_drain_amount(
+                    dummy_amount_sat,
+                    &lbtc_pair,
+                    drain_amount_sat,
+                );
+                lbtc_pair
+                    .limits
+                    .within(invoice_amount_sat)
+                    .map_err(|e| LnUrlPayError::Generic { err: e.message() })?;
+                // Validate if we can actually drain the wallet with a swap
+                let pair_fees_sat = lbtc_pair.fees.total(invoice_amount_sat);
+                ensure_sdk!(
+                    invoice_amount_sat + pair_fees_sat == drain_amount_sat,
```

Comment:
In case this doesn't hold isn't it better to add 1 sat to the fees to match the drain amount?

Reply:
These are the pair fees that boltz deduct. The one option we have is to increase the lockup fee by 1 sat in this case, so we need to adjust the default fee rate when creating the lockup for send_payment